src/DynamoPackages/PackageLoader.cs

```diff
@@ -386,16 +389,28 @@ public void LoadPackages(IEnumerable<Package> packages)
         }
 
         /// <summary>
-        /// Scans and compiles any new packages contained in new package paths that are added.
-        /// Note: This method will still reload ALL packages in the VM even those that are not new.
+        /// This method is called when custom nodes and packages need to be reloaded if there are new package paths.
         /// </summary>
         /// <param name="loadPackageParams"></param>
-        private void LoadNewPackages(LoadPackageParams loadPackageParams)
+        /// <param name="customNodeManager"></param>
+        public void LoadCustomNodesAndPackages(LoadPackageParams loadPackageParams, CustomNodeManager customNodeManager)
         {
+            foreach (var path in loadPackageParams.Preferences.CustomPackageFolders)
+            {
+                // Append the definitions subdirectory for custom nodes.
+                var dir = path == DynamoModel.BuiltInPackagesToken ? PathManager.BuiltinPackagesDirectory : path;
+                dir = TransformPath(dir, PathManager.DefinitionsDirectoryName);
+
+                customNodeManager.AddUninitializedCustomNodesInPath(dir, false, false);
+            }
+
             foreach (var path in loadPackageParams.NewPaths)
             {
-                var packagesDirectory = pathManager.PackagesDirectories.Where(x => x.StartsWith(path)).FirstOrDefault();
-                ScanPackageDirectories(packagesDirectory, loadPackageParams.Preferences);
+                var packageDirs = pathManager.PackagesDirectories.Where(x => x.StartsWith(path));
```

Comment:
so this has been improved so more than one path can be added?

Reply:
No, not really. We still only scan one path.
I really don't remember what I was trying to solve by removing the FIrstOrDefault.
I'll put that back